packages/devextreme/js/__internal/ui/m_multi_view.ts

```diff
@@ -142,14 +142,51 @@ const MultiView = CollectionWidget.inherit({
     this._initSwipeable();
   },
 
+  _ensureSelectedItemIsVisible(): void {
+    const { items } = this.option();
+    let currentSelectedIndex = this.option('selectedIndex');
+
+    if (currentSelectedIndex >= items.length) {
+      currentSelectedIndex = items.length - 1;
+      this.option('selectedIndex', currentSelectedIndex);
+    }
+
+    if (this._isItemVisible(currentSelectedIndex)) {
+      return;
+    }
+
+    const allItemsHidden = items.every((_, index) => !this._isItemVisible(index));
+    const isIndexOverflowed = (this._itemsCount() - 1) < currentSelectedIndex;
```

Comment:
[excess not working code]
it's always flow because of the if [condition above above](https://github.com/DevExpress/DevExtreme/pull/28145/files#diff-b95a070f9f0a64ed59233db53b817dce631e45fc0a7e5c12ff624c54ffd8b93aR149)